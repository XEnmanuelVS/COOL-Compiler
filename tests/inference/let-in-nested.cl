class Main {
    func(): AUTO_TYPE {
        let x: AUTO_TYPE <- 3, y: AUTO_TYPE <- x in
            let v: AUTO_TYPE <- y, w: AUTO_TYPE <- v in
                v
    };
};

