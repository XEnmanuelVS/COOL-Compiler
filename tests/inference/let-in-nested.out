\__ProgramNode [<class> ... <class>]
  \__ClassDeclarationNode: class Main  { <feature> ... <feature> }
    \__FuncDeclarationNode: func() : Int -> <body>
      \__LetNode:  let
       \__VarDeclarationNode: x: Int <-
        \__IntegerNode: 3
      \__VarDeclarationNode: y: Int <-
        \__VariableNode: x
        in
          \__LetNode:  let
           \__VarDeclarationNode: v: Int <-
            \__VariableNode: y
          \__VarDeclarationNode: w: Int <-
            \__VariableNode: v
            in
              \__VariableNode: v

